### PR TITLE
npush: init at 0.7

### PR DESCRIPTION
**Pull request by AndersonTorres**
Also, include a runner script to help copying level files before running the
game.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/npush/run.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+         echo "Directory ./levels not found; copying the official levelset"
+         mkdir -p ./levels
+         cp ${npush}/share/npush/levels/* levels/
+         chmod 644 levels/*
+      fi
+
+      exec "${npush}/bin/npush"
+    '';
+    executable = true;
+    destination = "/bin/run-npush";
```
Comment (legendofmiracles) [inline pkgs/games/npush/run.nix:29]: These shouldn't be needed if you use the more idiomatic function

File: pkgs/games/npush/run.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+, symlinkJoin
+, writeTextFile
+, npush
+}:
+
+let
+  runScript = writeTextFile {
+    name = "run-npush";
+    text = ''
+      #!${runtimeShell}
```
[the remark this answers is not in the document]
Comment (IvarWithoutBones) [reply pkgs/games/npush/run.nix:11]: See https://github.com/NixOS/nixpkgs/pull/155263#pullrequestreview-855754612 for the reasoning

File: pkgs/games/npush/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+, fetchurl
+, ncurses
+}:
+
+stdenv.mkDerivation rec {
+  pname = "npush";
+  version = "0.7";
+
+  src = fetchurl {
+    url = "https://downloads.sourceforge.net/project/npush/${pname}/${version}/${pname}-${version}.tgz";
```
Comment (sikmir) [inline pkgs/games/npush/default.nix:12]:
```suggestion
    url = "mirror://sourceforge/project/npush/npush/${version}/npush-${version}.tgz";
```

File: pkgs/games/npush/run.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/games/npush/run.nix:11]: As I have seen, this is not the expected syntax of `writeShellScriptBin`. 

I will fix this nonetheless

**Pull request merged**